endpoints/zones/src/main.cpp

```diff
@@ -9,78 +9,468 @@
+						}}
+					}.dump();
+					// clang-format on
+				}
+				catch (const std::exception& e) {
+					logging::error("{}: {}", fn, e.what());
+					res.result(http::status::internal_server_error);
+				}
+
+				res.prepare_payload();
+
+				return _sess_ptr->send(std::move(res));
+			});
+	} // op_remove
+
+	IRODS_HTTP_API_ENDPOINT_OPERATION_SIGNATURE(op_modify)
+	{
+		auto result = irods::http::resolve_client_identity(_req);
+		if (result.response) {
+			return _sess_ptr->send(std::move(*result.response));
+		}
+
+		const auto client_info = result.client_info;
+
+		irods::http::globals::background_task(
+			[fn = __func__, client_info, _sess_ptr, _req = std::move(_req), _args = std::move(_args)] {
+				logging::info("{}: client_info.username = [{}]", fn, client_info.username);
+
+				http::response<http::string_body> res{http::status::ok, _req.version()};
+				res.set(http::field::server, irods::http::version::server_name);
+				res.set(http::field::content_type, "application/json");
+				res.keep_alive(_req.keep_alive());
+
+				try {
+					const auto name_iter = _args.find("name");
+					if (name_iter == std::end(_args)) {
+						logging::error("{}: Missing [name] parameter.", fn);
+						return _sess_ptr->send(irods::http::fail(res, http::status::bad_request));
+					}
 
-			BytesBuf* bbuf{};
-			irods::at_scope_exit free_bbuf{[&bbuf] { freeBBuf(bbuf); }};
+					const auto property_iter = _args.find("property");
+					if (property_iter == std::end(_args)) {
+						logging::error("{}: Missing [property] parameter.", fn);
+						return _sess_ptr->send(irods::http::fail(res, http::status::bad_request));
+					}
 
-			{
-				auto conn = irods::get_connection(client_info.username);
+					const auto value_iter = _args.find("value");
+					if (value_iter == std::end(_args)) {
+						logging::error("{}: Missing [value] parameter.", fn);
+						return _sess_ptr->send(irods::http::fail(res, http::status::bad_request));
+					}
 
-				if (const auto ec = rcZoneReport(static_cast<RcComm*>(conn), &bbuf); ec != 0) {
-					logging::error("{}: rcZoneReport error: [{}]", fn, ec);
-					return _sess_ptr->send(irods::http::fail(status_type::bad_request));
+					auto conn = irods::get_connection(client_info.username);
+
+					if (property_iter->second == "name") {
+						adm::client::modify_zone(conn, name_iter->second, adm::zone_name_property{value_iter->second});
+					}
+					else if (property_iter->second == "connection_info") {
+						adm::client::modify_zone(
+							conn, name_iter->second, adm::connection_info_property{value_iter->second});
+					}
+					else if (property_iter->second == "comment") {
+						adm::client::modify_zone(conn, name_iter->second, adm::comment_property{value_iter->second});
+					}
+					else if (property_iter->second == "acl") {
+						// The string referenced by the value iterator is expected to have the following structure:
+					    //
+					    //     <acl>:<user>
+					    //
```

Comment:
Still need to think about the following:
- Should this be exposed at all? _(Maybe. It is a valid operation.)_
- Should consider changing how the input argument is provided (i.e. `<acl>:<user>`)

Reply:
After some experimentation with `iadmin modzonecollacl`, adding a dedicated operation for it makes the most sense.

Removing.

Reply:
Done.